apps/settings/js/panels/call/panel.js

```diff
@@ -0,0 +1,466 @@
+  var SettingsPanel = require('modules/settings_panel');
+  var SettingsCache = require('modules/settings_cache');
+  var SettingsService = require('modules/settings_service');
+  var DsdsSettings = require('dsds_settings');
+  var DialogService = require('modules/dialog_service');
+
+  return function ctor_call_panel() {
+    return SettingsPanel({
+      onInit: function(panel, options) {
+        this._cardIndex = options.cardIndex || 0;
+        this._conns = window.navigator.mozMobileConnections;
+        this._conn = this._conns[this._cardIndex];
+
+        this._elements = {};
+        this._elements.panel = panel;
+        this._elements.header = panel.querySelector('gaia-header');
+        this._elements.rilCallerId = panel.querySelector('#ril-callerId');
+        this._elements.fdnDesc = panel.querySelector('#fdnSettings-desc');
+        this._elements.alertPanel = panel.querySelector('#call .cw-alert');
+        this._elements.alertLabel =
+          panel.querySelector('#menuItem-callWaiting .alert-label');
+        this._elements.alertPanelSetButton =
+          this._elements.alertPanel.querySelector('.cw-alert-set');
+        this._elements.alertPanelConfirmInput =
+          this._elements.alertPanel.querySelector(
+            '.cw-alert-checkbox-label input');
+        this._elements.alertPanelCancelButton =
+          this._elements.alertPanel.querySelector('.cw-alert-cancel');
+        this._elements.fdnItem = panel.querySelector('#menuItem-callFdn');
+        this._elements.callWaitingItem =
+          panel.querySelector('#menuItem-callWaiting');
+        this._elements.callWaitingInput =
+          this._elements.callWaitingItem.querySelector('.checkbox-label input');
+        this._elements.callForwardingItem =
+          panel.querySelector('#menuItem-callForwarding');
+        this._elements.callForwardingAnchor =
+          this._elements.callForwardingItem.querySelector('a');
+        this._elements.callBarringItem =
+          panel.querySelector('#menuItem-callBarring');
+        this._elements.callerIdItem =
+          panel.querySelector('#menuItem-callerId');
+        this._elements.voiceMailItem =
+          panel.querySelector('.menuItem-voicemail');
+        this._elements.voiceMailDesc =
+          panel.querySelector('#voiceMail-desc');
+        this._elements.voicePrivacyItem =
+          panel.querySelector('#menuItem-voicePrivacyMode');
+        this._elements.voicePrivacyInput =
+          this._elements.voicePrivacyItem.querySelector('input');
+
+        // Set the navigation correctly when on a multi ICC card device.
+        if (DsdsSettings.getNumberOfIccSlots() > 1) {
+          this._elements.header.setAttribute('data-href', '#call-iccs');
+        }
+
+        this._bindPanelReadyEvent();
+        this._bindVisibilityChangeEvent();
+        this._bindCallForwardingMenuItemClickEvent();
+
+        this._addVoiceTypeChangeListeners();
+
+        this._initCallerId();
```

Comment:
Note that all functions relying on a specific sim should be called in `onBeforeShow` instead of `onInit`. `onInit` is only called once.